### DISCUSSION TITLE
API playback interruption

### DISCUSSION THREAD

**Discussion by raffa0001** [erew123/alltalk_tts#281] [Q&A]
I'm trying to integrate alltalk with another program and i need to add a command to interrupt playback when needed through the API suite, i noticed in the documentation that there's no such option.
Am i missing something?

**Comment by RenNagasaki** [comment]
There should be this call: /api/stop-generation

**Reply by RenNagasaki** [reply]
![grafik](https://github.com/user-attachments/assets/56a17591-c24e-4127-8578-b82539dd5812)

**Reply by raffa0001** [reply]
Excuse me but i can't find this, where is it?
In the standalone server documentation there's no such thing, is it from the v2 beta?
I found the /api/ endpoints in tts_server.py but i can't seem to find its documentation

**Reply by RenNagasaki** [reply]
Oh, jeah. The description is from the v2 Beta. But the endpoint should still work in v1 iirc

**Reply by raffa0001** [reply]
Yeah unfortunately though that command doesn't stop playback, it just interrupts generation.
Let's say i want to stop the TTS mid sentence, it just won't by sending stop-generation, i need it to interrupt the server playback rather than generation.

**Comment by erew123** [comment]
@raffa0001 

Do you mean you are using `autoplay=true` when generating TTS to play the generated audio file at the console/terminal where AllTalk is running, and that is where you want to be able to stop playback etc?

**Reply by raffa0001** [reply]
Yeah that's right, i'd like to be able to interrupt the terminal's playback, it'd probably be easier though simply working with the generated wav file turning autoplay off rather, for what i'm trying to do.

**Reply by erew123** [reply]
@raffa0001 It will be yeah. There is currently no code or API in place to control the playback at the console. So if you do have a local audio player within your code, using the "output_file_url" from the JSON return will be your best bet https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-tts-generation-response

Please note there is a difference between v1 and v2 of AllTalk with path to the "output_file_url", which is discussed in the documentation of v2 and also here https://github.com/erew123/alltalk_tts/issues/166

Thanks
